node_follower/src/follower_service.rs

```diff
@@ -24,12 +20,15 @@ pub struct FollowerService {
 
 #[async_trait::async_trait]
 impl GatewayInfoResolver for FollowerService {
-    async fn resolve_gateway_info(&mut self, address: &PublicKey) -> Result<FollowerGatewayResp> {
+    async fn resolve_gateway_info(&mut self, address: &PublicKey) -> Result<GatewayInfo> {
         let req = FollowerGatewayReqV1 {
             address: address.to_vec(),
         };
         let res = self.client.find_gateway(req).await?.into_inner();
-        Ok(res.try_into()?)
+        match res.result {
+            Some(GatewayResult::Info(gateway_info)) => Ok(gateway_info.try_into()?),
+            _ => Err(Error::GatewayNotFound(format!("{address}"))),
```

Comment:
Should you not explicitly match on `follower_error` here, rather than default all errors to gateway not found ?

Reply:
Expediency at the moment; since there’s only the one `follower_error` alternative result at the moment.